content/articles/2019-05-vroom-1-0-0.Rmarkdown

```diff
@@ -0,0 +1,450 @@
+vroom_write(flights, "flights.tsv")
+
+# These are taken from the vroom vignette
+pretty_sec <- function(x) {
+  x[!is.na(x)] <- prettyunits::pretty_sec(x[!is.na(x)])
+  x
+}
+
+library(forcats)
+library(dplyr)
+library(ggplot2)
+library(tidyr)
+read_benchmark <- function(file, desc) {
+  vroom::vroom(file, col_types = c("cccdddd")) %>%
+    filter(type == "real", op != "setup") %>%
+    mutate(
+      package = fct_reorder(sub("_", "\n", package), time, sum),
+      op = factor(op, desc)
+    )
+}
+
+generate_subtitle <- function(data) {
+  rows <- scales::comma(data$rows[[1]])
+  cols <- scales::comma(data$cols[[1]])
+  size <- fs::fs_bytes(data$size[[1]])
+  glue::glue("{rows} x {cols} - {size}B")
+}
+
+plot_benchmark <- function(data, title) {
+
+  subtitle <- generate_subtitle(data)
+
+  data %>%
+    ggplot() +
+    geom_bar(aes(x = package, y = time, fill = op, group = package), stat = "identity") +
+    scale_fill_brewer(type = "qual", palette = "Set2") +
+    scale_y_continuous(labels = function(x) format(bench::as_bench_time(x))) +
+    theme_grey(base_size = 18) +
+    theme(legend.position = "bottom") +
+    coord_flip() +
+    labs(title = title, subtitle = subtitle, x = NULL, y = NULL, fill = NULL)
+}
+
+make_table <- function(data) {
+  times <- data %>%
+    group_by(package, op) %>%
+    filter(type == "real") %>%
+    tally(wt = time) %>%
+    spread(op, n) %>%
+    mutate(total = sum(read, print, head, tail, sample, filter, aggregate))
+
+  times %>%
+    arrange(desc(total)) %>%
+    mutate_if(is.numeric, pretty_sec) %>%
+    knitr::kable(digits = 2, align = "r", format = "markdown")
+}
+
+desc <- c("setup", "read", "print", "head", "tail", "sample", "filter", "aggregate")
+```
+
+<html>
+<style>
+h2 code {
+    font-size: 1em;
+}
+</style>
+</html>
+
+I'm excited to announce that [vroom 1.0.0](http://vroom.r-lib.org) is now
+available on CRAN!
+
+vroom reads rectangular data, such as comma separated
+(csv), tab separated (tsv) or fixed width files (fwf) into R.
+
+It performs the
+same function as packages like [readr](http://readr.r-lib.org),
+[data.table](http://r-datatable.com) or the `read.csv()`
+function, but for many datasets vroom can read them much, much
+faster (hence the name).
+
+The main reason vroom can be faster is because character data is read from the
+file lazily; you only pay for the data you use. This lazy access is done
+automatically, so no changes to your R data manipulation code are needed.
+
+vroom also provides efficient multi-threaded writing that is multiple times
+faster on most inputs than the readr writer.
+
+Install vroom with:
+
+```{r, eval = FALSE}
+install.packages("vroom")
+```
+
+The best way to get acquainted with the package is the [getting
+started](http://vroom.r-lib.org/articles/vroom.html) vignette.
+
+## Reading delimited files
+
+Compared to readr, the first difference you may note is you use only one
+function to read the files, `vroom()`. This is because `vroom()` guesses the
```

Comment:
Link to function reference on first mention. [`vroom()`](http://vroom.r-lib.org/reference/vroom.html)